### ISSUE TITLE
It is possible to submit /join form too often

### ISSUE THREAD

**Issue by amihaiemil** [zerocracy/farm#653]
As discussed, the /join form should be submittable only once a week, to avoid white spamming.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@amihaiemil but all forms require login. You can't fill them out unless you are logged in. Can you imagine a GitHub user spamming us through those forms? What for? And we can easily block him/her.

Maybe we should allow `/join` form to be submitted only once per, say, 7 days. In order to prevent "white" spam, when an honest user really wants to be invited, but the profile is not strong enough and nobody wants to invite him/her. Make sense?

**Comment by amihaiemil** [comment]
@yegor256 yes, allowing ``/join`` only once in a while sounds better. 

Even if user is logged in, I could easily make a Github false account and start spamming. We will detect it eventually, but it's enough if one makes a DDOS once a month and 0crat is down for 1 day until someone realizes it :-?

**Comment by yegor256** [comment]
@amihaiemil but we have only two forms: `/join` and `/rfp`. Join will allow you to spam only once a week. `/rfp` requires you to pay $16 every time you submit it. So, nothing to be afraid of :)

**Comment by yegor256** [comment]
@amihaiemil let's change the description of the ticket or close this one and create a new one. The bug is that it's possible to submit `/join` too frequently. Should be once a week.

**Comment by amihaiemil** [comment]
@yegor256 done

**Comment by yegor256** [comment]
@amihaiemil the solution would be to add `<applied/>` element to `people.xml` with date/time when the request has been received. Then, we should check that value when a new request comes in.

**Comment by 0crat** [comment]
Job [#653](https://github.com/zerocracy/farm/issues/653) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+1145](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
The job [#653](https://github.com/zerocracy/farm/issues/653) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/691742898). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by yegor256** [comment]
@g4s8 I think you are using a wrong strategy while working with multiple repositories. When you have a ticket that needs changes in multiple places, you should submit tickets to other repositories, explaining there what should be done. Then, someone (!) will fix them, release new versions and you will continue here.

**Comment by g4s8** [comment]
@yegor256 got it, I will do it for new tickets

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 20days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat waiting for zerocracy/datum release

**Comment by 0crat** [comment]
> @0crat waiting for zerocracy/datum release ([here](https://github.com/zerocracy/farm/issues/653#issuecomment-377547438))

@g4s8 The impediment for [#653](https://github.com/zerocracy/farm/issues/653) was registered successfully by @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0pdd** [comment]
@amihaiemil the puzzle [#956](https://github.com/zerocracy/farm/issues/956) is still not solved.

**Comment by g4s8** [comment]
@amihaiemil PR #881 was merged, todo #956 was submitted, can we close?

**Comment by amihaiemil** [comment]
@g4s8 yeah, thanks :)

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@d20feeb0-0ee6-4a68-b678-bb443d7a88c9, thread: BkParallel-1
com.mongodb.connection.InternalStreamConnection[457] com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
java.net.SocketInputStream[-2] java.net.SocketTimeoutException: Read timed out
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@d20feeb0-0ee6-4a68-b678-bb443d7a88c9, thread: AsyncFlush-7
com.zerocracy.farm.reactive.StkGroovy[125] java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/in/orders/close_job.groovy
sun.reflect.NativeMethodAccessorImpl[-2] java.lang.reflect.InvocationTargetException: null
com.mongodb.connection.InternalStreamConnection[457] com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
java.net.SocketInputStream[-2] java.net.SocketTimeoutException: Read timed out
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @g4s8[/z](https://www.0crat.com/u/g4s8), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#653](https://github.com/zerocracy/farm/issues/653) is now out of scope

**Comment by 0crat** [comment]
The job is not in WBS, won't close the order

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@amihaiemil the only puzzle [#956](https://github.com/zerocracy/farm/issues/956) is solved here.
